### PR TITLE
Added normalisation method for cache keys, updated and added debug logs

### PR DESCRIPTION
**Pull request by thomasdigby**
#### What have you done
Added a terrible piece of regex to normalise the cache keys. Applied to `siteUrl` in API handler as it was sometimes doubling up the first `/`.

Also added and updated a bunch of the debug logs to improve their usefulness.

#### Why have you done it
#206 

#### Testing carried out to prevent breaking changes
Added tests for setting cache items and retrieving them

### CODE REVIEW

File: src/server/handle-api.js [hunk 1]
```diff
@@ -15,26 +15,35 @@ export default ({ server, config }) => {
     path: '/api/v1/{query*}',
     handler: (req, reply) => {
 
-      const base = `${config.siteUrl}/wp-json/wp/v2`
+      const base = `${normalizePath(config.siteUrl)}/wp-json/wp/v2`
```
Comment (mathewhawley) [inline src/server/handle-api.js]: i wonder if node's `path.normalize` could do this work here? pass the full string in? not 100% but could be a good way to do this

File: src/server/handle-api.js [hunk 2]
```diff
@@ -15,26 +15,35 @@ export default ({ server, config }) => {
...
       const path = `${req.params.query}${req.url.search}`
       const remote = `${base}/${path}`
+      const cacheKey = normalizePath(path)
```
Comment (mathewhawley) [inline src/server/handle-api.js]: I wonder if `normalizePath` is the right name to use here? Am i correct in thinking it just strips leading and trailing slashes? Wondering if it'd help to make name tie more in to that?

File: src/server/handle-api.js (hunk 1 above)
Comment (thomasdigby) [inline src/server/handle-api.js]: Yep in this use-case you're probably right

File: src/server/handle-api.js (hunk 2 above)
Comment (thomasdigby) [inline src/server/handle-api.js]: I'll rename it 👍

File: src/server/handle-api.js (hunk 1 above)
Comment (thomasdigby) [inline src/server/handle-api.js]: Actually reading through the docs makes me think this is about relative file/folder paths? Rather than URLs

Comment (mathewhawley) [inline src/server/handle-api.js]: true – up to you – it looks like you could use it as you're not dealing with protocols etc, just a straight path?

**Pull request merged**